src/archivist/services/index.js

```diff
@@ -4,261 +4,284 @@ import { pathToFileURL } from 'url';
+    loadServiceHistory(filePaths.history),
+    loadServiceFiltersHistory(filePaths.filtersHistory, filePaths.filters),
+  ]);
 
-  Object.keys(serviceDeclaration.terms).forEach(termsType => {
+  Object.entries(serviceDeclaration.terms).forEach(([ termsType, declaration ]) => {
     serviceHistory[termsType] = serviceHistory[termsType] || [];
-    serviceHistory[termsType].push(serviceDeclaration.terms[termsType]);
+    serviceHistory[termsType].push(declaration);
   });
 
   sortHistory(serviceHistory);
+  sortHistory(serviceFiltersHistory);
 
-  if (await fileExists(serviceFiltersHistoryFileName)) {
-    serviceFiltersHistoryModule = await import(pathToFileURL(serviceFiltersHistoryFileName));
-    Object.keys(serviceFiltersHistoryModule).forEach(filterName => {
-      serviceFiltersHistory[filterName] = serviceFiltersHistoryModule[filterName];
-    });
+  return { declarations: serviceHistory, filters: serviceFiltersHistory };
+}
+
+function getHistoryFilePaths(serviceId) {
+  const basePath = path.join(declarationsPath, serviceId);
+
+  return {
+    history: `${basePath}${HISTORY_SUFFIX}${JSON_EXT}`,
+    filters: `${basePath}${FILTERS_SUFFIX}${JS_EXT}`,
+    filtersHistory: `${basePath}${FILTERS_SUFFIX}${HISTORY_SUFFIX}${JS_EXT}`,
+  };
+}
+
+async function loadServiceHistory(historyFilePath) {
+  if (!(await fileExists(historyFilePath))) return {};
+
+  try {
+    return JSON.parse(await fs.readFile(historyFilePath));
+  } catch (error) {
+    const fileName = path.basename(historyFilePath, JSON_EXT);
+
+    throw new Error(`The "${fileName}" service declaration is malformed and cannot be parsed in ${historyFilePath}`);
   }
+}
 
-  if (await fileExists(serviceFiltersFileName)) {
-    const serviceFilters = await import(pathToFileURL(serviceFiltersFileName));
+async function loadServiceFiltersHistory(filtersHistoryPath, filtersPath) {
+  const filtersHistory = {};
 
-    Object.keys(serviceFilters).forEach(filterName => {
-      serviceFiltersHistory[filterName] = serviceFiltersHistory[filterName] || [];
-      serviceFiltersHistory[filterName].push({ filter: serviceFilters[filterName] });
-    });
+  if (await fileExists(filtersHistoryPath)) {
+    const historyModule = await import(pathToFileURL(filtersHistoryPath));
+
+    Object.assign(filtersHistory, historyModule);
   }
 
-  sortHistory(serviceFiltersHistory);
+  if (await fileExists(filtersPath)) {
+    const filtersModule = await import(pathToFileURL(filtersPath));
 
-  return {
-    declarations: serviceHistory || {},
-    filters: serviceFiltersHistory || {},
-  };
+    Object.entries(filtersModule).forEach(([ filterName, filter ]) => {
+      filtersHistory[filterName] = filtersHistory[filterName] || [];
+      filtersHistory[filterName].push({ filter });
+    });
+  }
+
+  return filtersHistory;
 }
 
 async function fileExists(filePath) {
   try {
     await fs.access(filePath);
 
     return true;
-  } catch (error) {
-    if (error.code === 'ENOENT') {
```

Comment:
Why was this removed? It reads like a safety mechanism to avoid being too broad about silencing errors, and I could not find a rationale for broadening in 2e69c40ccf5866654f9a555635d4562f9ac3fc5b.